File: DP/javascript/1932.js
const input = require("fs").readFileSync("/dev/stdin").toString().split("\n");
const n = Number(input[0]);
const dp = [];
const arr = [];

const calculateNum = (arr, r, c) => {
  if (r === 0) {
    return arr[r][c];
  }
  if (c === 0) {
    return arr[r][c] + dp[r - 1][c];
  }
  if (r === c) {
    return arr[r][c] + dp[r - 1][c - 1];
  }
  return arr[r][c] + Math.max(dp[r - 1][c - 1], dp[r - 1][c]);
};

for (let i = 1; i <= n; i++) {
  dp.push(Array(i).fill(false));
  arr.push(input[i].split(" ").map(Number));
}

for (let i = 0; i < n; i++) {
  for (let j = 0; j < i + 1; j++) {
    dp[i][j] = calculateNum(arr, i, j);
  }
}

console.log(Math.max(...dp[n - 1]));

File: DP/python/1932.py
import sys
input = sys.stdin.readline
n = int(input())
data = [list(map(int,input().split())) for _ in range(n)]
dp = [[0 for _ in range(i+1)] for i in range(n)]
dp[0][0] = data[0][0]
for i in range(1,n):
    dp[i][0] = dp[i-1][0] + data[i][0]
    for j in range(1,i):
        dp[i][j] = max(dp[i-1][j-1],dp[i-1][j])+data[i][j]
    dp[i][-1] = dp[i-1][-1] + data[i][-1]
answer = max(dp[-1])
print(answer)